alerts/lib/alerttask.py

```diff
@@ -109,56 +109,86 @@ def __init__(self):
         # We want to select all event indices
         # and filter out the window based on timestamp
         # from the search query
-        self.event_indices = ['events-*']
+        self.event_indices = ["events-*"]
 
     def classname(self):
         return self.__class__.__name__
 
     @property
     def log(self):
-        return get_task_logger('%s.%s' % (__name__, self.alert_name))
+        return get_task_logger("%s.%s" % (__name__, self.alert_name))
 
     def parse_config(self, config_filename, config_keys):
         myparser = OptionParser()
         self.config = None
         (self.config, args) = myparser.parse_args([])
         for config_key in config_keys:
-            temp_value = getConfig(config_key, '', config_filename)
+            temp_value = getConfig(config_key, "", config_filename)
             setattr(self.config, config_key, temp_value)
 
+    def _discover_task_exchange(self):
+        """Use configuration information to understand the message queue protocol.
+        return: amqp, sqs
+        """
+        return getConfig("mqprotocol", "amqp", None)
+
+    def __build_conn_string(self):
+        exchange_protocol = self._discover_task_exchange()
+        if exchange_protocol == "amqp":
+            connString = "amqp://{0}:{1}@{2}:{3}//".format(
+                RABBITMQ["mquser"],
+                RABBITMQ["mqpassword"],
+                RABBITMQ["mqserver"],
+                RABBITMQ["mqport"],
+            )
+            return connString
+        elif exchange_protocol == "sqs":
+            connString = "sqs://{}".format(getConfig("alertSqsQueueUrl", None, None))
+            if connString:
+                connString = connString.replace('https://','')
+            return connString
+
     def _configureKombu(self):
         """
-        Configure kombu for rabbitmq
+        Configure kombu for amqp or sqs
         """
         try:
-            connString = 'amqp://{0}:{1}@{2}:{3}//'.format(
-                RABBITMQ['mquser'],
-                RABBITMQ['mqpassword'],
-                RABBITMQ['mqserver'],
-                RABBITMQ['mqport'])
+            connString = self.__build_conn_string()
             self.mqConn = kombu.Connection(connString)
-
-            self.alertExchange = kombu.Exchange(
-                name=RABBITMQ['alertexchange'],
-                type='topic',
-                durable=True)
-            self.alertExchange(self.mqConn).declare()
-            alertQueue = kombu.Queue(RABBITMQ['alertqueue'], exchange=self.alertExchange)
+            if connString.find('sqs') == 0:
```

Comment:
Since you're creating the string on `__build_conn_string`, would it be better to verify if the string starts with "sqs://" instead of if `sqs` is in the connection string?